helios-core/src/main/kotlin/helios/instances/instancesJavaTime.kt

```diff
@@ -52,7 +53,11 @@ interface ZonedDateTimeDecoderInstance : Decoder<ZonedDateTime> {
   fun formatter(): DateTimeFormatter
 
   override fun decode(value: Json): Either<DecodingError, ZonedDateTime> =
-    value.asJsString().map { ZonedDateTime.parse(it.value, formatter()) }.toEither { StringDecodingError(value) }
+    value.asJsString().toEither { JsStringDecodingError(value) }.flatMap {
```

Comment:
We have a common pattern here. It'd be possible to have a `decodeString` in place? In that way we could have something like:

```kotlin
fun decodeString(value: Json): Either<DecodingError, String> = TODO()

override fun decode(value: Json): Either<DecodingError, ZonedDateTime> =
  decodeString(value).flatMap {
    Try { ZonedDateTime.parse(it.value, formatter()) }.toEither {
      DateDecodingError(value, formatter().some())
    }
  }
```